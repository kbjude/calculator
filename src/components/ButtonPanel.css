.row1,
.row2,
.row3
.row4
.row5 {
    height: 100px;
}

button {
    height: 100px;
}

.button-0 {
    width: 22.4px !important;
}

.button-panel {
    width: 700px;
}