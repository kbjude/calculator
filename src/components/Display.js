const Display = (props) => {
    return (
        <input type="text" value={props.text} />
    )
};