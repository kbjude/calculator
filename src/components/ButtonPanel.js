import PropTypes from 'prop-types';
import Button from './Button';
import './ButtonPanel.css';

const ButtonPanel = props => {
  const handleClick = buttonName => {
    props.clickHandler(buttonName);
  };

  return (
    <div className="button-panel">
      <h1> Calculator</h1>
      <>
        <div className="row1">
          <Button name="AC" clickHandler={handleClick} />
          <Button name="+/-" clickHandler={handleClick} />
          <Button name="%" clickHandler={handleClick} />
          <Button name="รท" clickHandler={handleClick} />
        </div>
        <div className="row2">
          <Button name="7" clickHandler={handleClick} />
          <Button name="8" clickHandler={handleClick} />
          <Button name="9" clickHandler={handleClick} />
          <Button name="x" clickHandler={handleClick} />
        </div>
        <div className="row3">
          <Button name="4" clickHandler={handleClick} />
          <Button name="5" clickHandler={handleClick} />
          <Button name="6" clickHandler={handleClick} />
          <Button name="-" clickHandler={handleClick} />
        </div>
        <div className="row4">
          <Button name="1" clickHandler={handleClick} />
          <Button name="2" clickHandler={handleClick} />
          <Button name="3" clickHandler={handleClick} />
          <Button name="+" clickHandler={handleClick} />
        </div>
        <div className="row5">
          <Button name="0" className="button-0" clickHandler={handleClick} />
          <Button name="." clickHandler={handleClick} />
          <Button name="=" clickHandler={handleClick} />
        </div>
      </>
    </div>
  );
};

ButtonPanel.propTypes = { clickHandler: PropTypes.func.isRequired };
export default ButtonPanel;
