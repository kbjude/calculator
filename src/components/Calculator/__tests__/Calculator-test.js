import Display from './../../Display'
import { string } from 'prop-types';
import { ButtonPanel } from './../../ButtonPanel';

describe('Calculator', () => {
    it("passing test", () => {
        expect(true).toBeTruthy();
    });

    it('Display result', () => {
        expect(Display.props).toBe(Display.result);
    })

    it('Checks for type', () => {
        expect(Display.propTypes).toString();
    })

})