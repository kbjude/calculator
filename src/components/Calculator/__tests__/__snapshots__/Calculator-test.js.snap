// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displaying out puts 1`] = `<calculate />`;
